plugins/techdocs/config.d.ts

```diff
@@ -32,5 +32,19 @@ export interface Config {
      * @deprecated
      */
     requestUrl?: string;
+
+    /**
+     * Alternate hostnames to identify Git hosting type for feedback links for TechDocs pages
+     */
+    feedback?: {
```

Comment:
What do you think of something like...

```typescript
export interface Config {
  techdocs: {
    /**
     * Used to identify source code or issue system for feedback links for TechDocs pages
     */
    feedback?: {
      /**
       * The host name of the target system (e.g. ghe.example.com)
       * @visibility frontend
       */
      hostName?: string;
      /**
       * Name of the target system (e.g. github)
       * @visibility frontend
       */
      provider?: string;
    };
  };
}
```

Would prevent invalid configs (e.g. specifying both `techdocs.feedback.github` and `techdocs.feedback.gitlab`), as well as keep the config schema from growing with more supported providers.

Reply:
Specifying multiple feedback hostnames still works, it just uses the first match available. This is in case of documents coming from different providers needing to be displayed in the same Backstage instance.

Reply:
🤔 &nbsp;Oh, interesting.  I guess I hadn't fully thought through the prospect that one's catalog might be made up of entities whose code is stored across multiple source code providers.

Given that's the case, maybe we should more closely mirror the way `integrations`' config is implemented (where each provider key is an array with details)?

Or actually, can we just pull these details from the `integrations` config itself?

Reply:
Will change to use `integrations`

Reply:
🤦‍♂️  As I was reading through / closing out past comments, I realized you @adrianke77  had made a comment about `integrations`...  Sorry for missing that.  To look at that one more time:

> _@backstage/integrations also won't be of help if the built HTML docs are from a Git source not specified in the integrations configuration, eg if they are built on external CI/CD._

I suppose the case where the `integrations` config won't be populated will be somewhat uncommon?  Even if one sets up TechDocs in the recommended way (with publishing via CI/CD), it's likely you still have one or more integrations configured for other features in Backstage (e.g. for pulling component yaml definitions from the Catalog, or for generating new components via the Scaffolder).

Reply:
Yeah will ignore that edge case on the grounds that it seems quite unlikely, for now.